### DISCUSSION TITLE
can duckdb create a table in memory when open a db file?

### DISCUSSION THREAD

**Discussion by l1t1** [duckdb/duckdb#6939] [General]
like some dbms' memory engine

**Comment by Alex-Monahan** [comment]
Howdy!
If you use the `temp` keyword, the table is stored in memory, even if the database is backed by a file on disk:
```sql
Create temp table my_table as
Select * from ...
```

**Comment by l1t1** [comment]
thanks. 
https://duckdb.org/docs/sql/statements/create_table#temporary-tables
